samza-core/src/main/java/org/apache/samza/coordinator/stream/CoordinatorStreamSystemConsumer.java

```diff
@@ -124,6 +124,7 @@ public void start() {
     }
     log.info("Starting coordinator stream system consumer.");
     systemConsumer.start();
+    systemAdmin.start();
```

Comment:
Should be started and stopped by the object that passed in this SystemAdmin, not this object.

Reply:
`systemAdmin` is started here because the `systemAdmin` is instantiated in `CoordinatorStreamSystemFactory.getCoordinatorStreamSystemConsumer()`. The caller of this method actually does not know about SystemAdmin. How about I change the constructor of `CoordinatorStreamSystemConsumer` to be `CoordinatorStreamSystemConsumer(config: Config, registry: MetricsRegistry)` and remove the existing class `CoordinatorStreamSystemFactory`?

Reply:
CoordinatorStreamSystemConsumer constructor now takes config as the parameter.